### PR TITLE
Updated dagger dependency

### LINKED ISSUE #158: Upgrade dagger dependency
#### Problem description
We use [dagger](https://github.com/google/dagger) for injecting dependencies. The current version is 2.16. 
A new version of dagger 2 is also available (2.19).

### PR DESCRIPTION
**Pull request by iamvasuh**
Closes #158 

####What has been done to verify that this works as intended?

when I did the commit and ran the gradle sync in the android studio, it's not showing any error and also builds successfully thus verifying that it works as intended.

####Why is this the best possible solution? Were any other approaches considered?

Yes, this is the best possible solution.

####How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

As the dagger version is updated from 2.16 to 2.19 , users will get to use the latest versoin of the respective dagger-2 library. It's working fine also.
https://user-images.githubusercontent.com/42612183/53687851-09ad0700-3d60-11e9-822e-3aaee8c300af.jpeg

#### Before submitting this PR, please make sure you have:
- [ ] run `./gradlew pmd checkstyle lint findbugs` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/share/blob/master/share_app/src/main/assets/open_source_licenses.html).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -30,5 +30,5 @@ task clean(type: Delete) {
 ext {
     supportLibraryVersion = '27.1.1'
     butterknifeVersion = '8.8.1'
-    daggerVersion = '2.16'
-}
\ No newline at end of file
+    daggerVersion = '2.19'
+}
```
Comment (lakshyagupta21) [inline build.gradle]: Still, the same issue, remove this line change.

Comment (iamvasuh) [reply build.gradle]: that warning "no newline at end of file" is coming by default

Comment (shobhitagarwal1612) [reply build.gradle]: I think this is fine for now

Comment (iamvasuh) [reply build.gradle]: ok

**Pull request merged**